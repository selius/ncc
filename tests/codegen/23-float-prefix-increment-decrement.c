int main()
{
	float a;
	float b;

	a = 4.39;

	b = ++a;
	__print_float(a);
	__print_float(b);

	++a;
	b = ++a;
	__print_float(a);
	__print_float(b);

	a = -2.087;

	b = --a;
	__print_float(a);
	__print_float(b);

	--a;
	b = --a;
	__print_float(a);
	__print_float(b);

	return 0;
}
