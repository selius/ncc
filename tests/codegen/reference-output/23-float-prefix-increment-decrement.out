5.390000
5.390000
7.390000
7.390000
-3.087000
-3.087000
-5.087000
-5.087000
